packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -1,453 +1,17 @@
-    }
-  }
-
-  /**
-   * Stop spinning (clear any currently pending update and set spinning to false)
-   */
-  @autobind
-  private _stop() {
-    if (this._currentStepFunctionHandle >= 0) {
-      this._async.clearTimeout(this._currentStepFunctionHandle);
-      this._currentStepFunctionHandle = -1;
-    }
-
-    if (this._spinningByMouse || this.state.keyboardSpinDirection !== KeyboardSpinDirection.notSpinning) {
-      this._spinningByMouse = false;
-      this.setState({ keyboardSpinDirection: KeyboardSpinDirection.notSpinning });
-    }
-  }
-
-  /**
-   * Handle keydown on the text field. We need to update
-   * the value when up or down arrow are depressed
-   * @param event - the keyboardEvent that was fired
-   */
-  @autobind
-  private _handleKeyDown(event: React.KeyboardEvent<HTMLElement>) {
-    if (this.props.disabled) {
-      this._stop();
-
-      // eat the up and down arrow keys to keep the page from scrolling
-      if (event.which === KeyCodes.up || event.which === KeyCodes.down) {
-        event.preventDefault();
-        event.stopPropagation();
-      }
-
-      return;
-    }
-
-    let spinDirection = KeyboardSpinDirection.notSpinning;
-
-    if (event.which === KeyCodes.up) {
-
-      spinDirection = KeyboardSpinDirection.up;
-      this._updateValue(false /* shouldSpin */, this._initialStepDelay, this._onIncrement!);
-    } else if (event.which === KeyCodes.down) {
-
-      spinDirection = KeyboardSpinDirection.down;
-      this._updateValue(false /* shouldSpin */, this._initialStepDelay, this._onDecrement!);
-    } else if (event.which === KeyCodes.enter) {
-      event.currentTarget.blur();
-      this.focus();
-    } else if (event.which === KeyCodes.escape) {
-      if (this.state.value !== this._lastValidValue) {
-        this.setState({ value: this._lastValidValue });
-      }
-    }
-
-    // style the increment/decrement button to look active
-    // when the corresponding up/down arrow keys trigger a step
-    if (this.state.keyboardSpinDirection !== spinDirection) {
-      this.setState({ keyboardSpinDirection: spinDirection });
-    }
-  }
-
-  /**
-   * Make sure that we have stopped spinning on keyUp
-   * if the up or down arrow fired this event
-   * @param event stop spinning if we
-   */
-  @autobind
-  private _handleKeyUp(event: React.KeyboardEvent<HTMLElement>) {
-
-    if (this.props.disabled || event.which === KeyCodes.up || event.which === KeyCodes.down) {
-      this._stop();
-      return;
-    }
-  }
-
-  @autobind
-  private _onIncrementMouseDown() {
-    this._updateValue(true /* shouldSpin */, this._initialStepDelay, this._onIncrement!);
-  }
-
-  @autobind
-  private _onDecrementMouseDown() {
-    this._updateValue(true /* shouldSpin */, this._initialStepDelay, this._onDecrement!);
-  }
-}
+export const SpinButton = styled(
+  SpinButtonBase,
+  getStyles,
+  getArrowButtonStyles
```

Comment:
@dzearing  I'm a bit confused by the styling doc for customized sub component styling. In the guide, you mention passing in an object with multiple getStyles functions, but the styled function only seems to take in one getStyles function and a getProps function. If I pass in additional getStyles functions through as props, it gets bound to the base component, but it completely overrides the default getStyles for that subcomponent. If I understand correctly, that is not the desired behavior. Have I missed something here?

Reply:
I think that for this scenario, what we generally do is create also a base component for the subcomponent and use the styled function on it's base too (see: ColorPickerGridCellBase, ColorPickerGridCell), which would then merge the styles together (custom and base) for the subcomponent. The thing in this case is that the button component has not been converted to getStyles (which is itself a big change), so that will not work. We could create a wrapper component SpinButtonArrowButtonBase which returns the IconButton or just do the merge internally in the getUp/DownArrowButtonStyles function (not through styled) similar to what you did for Combobox.